### PR TITLE
BRS-1020: New Metrics FE changes

### PR DESCRIPTION
**Pull request by cameronpettit**
### Jira Ticket:
BRS-1020

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-1020

### Description:
Requires https://github.com/bcgov/parks-reso-api/pull/252.

## IMPORTANT:
There are now two very similar instances of metrics related services in the codebase:

 - Instances of `metric` refer to implementations for the OLD metrics services & systems.
 - Instances of `metrics` <- NOTE THE plural 's' refer to implementations of the NEW metrics system. 

References to the old implementations remain in the code base as their functionality is important for future tickets (BRS-1049) but should be phased out in a future ticket. 

Front end changes to go along with the new API changes for continuous metrics monitoring. Statistics for this iteration include Pass Breakdown by Status and Pass Activity by Day. Date range length is limited to a 1 year period maximum. 

Resolver will preselect 'All Parks' on first navigation to the metrics page, but user must select a date or date range to do an initial search of metrics. If a refinement of params is done on the page, the params will be saved if the user goes to a different part of the site and then comes back. 

Pass Activity By Day will only render if a single facility has been selected.

Pass Breakdown by status will only render if passes were found. 

The old `Export` button that generated a .csv of all passes in the system has been removed - it is expected that this button and its functionality will be reintroduced in https://bcparksdigital.atlassian.net/browse/BRS-1049 as the new metrics system is better suited to generate this csv. The functionality for the old export remains in the codebase as reference but should probably be removed as part of BRS-1049.

### CODE REVIEW

**Review by marklise — APPROVED** [review]
/w nit.

File: src/app/shared/utils/count-to.directive.ts [hunk 1]
```diff
@@ -103,6 +104,6 @@ export class CountToDirective implements OnDestroy, OnInit {
   }
 
   ngOnDestroy(): void {
-      this.ngUnsubscribe.complete();
+    this.ngUnsubscribe.complete();
```
Comment (marklise) [inline src/app/shared/utils/count-to.directive.ts:107]: I think we need a `.next()` before the `.complete()` here.

Comment (cameronpettit) [reply src/app/shared/utils/count-to.directive.ts:108]: fixed in `4242c7a`

**Comment by cameronpettit** [comment]
Noting the following for future benefit: 

The metrics filter is subscribed to park/facilities changes so that the park/facility picklists can be populated. When the autofetcher fires (every 5 mins) the picklists should be updated in case any new parks/facilities were added - however, when updating the select options for the picklist, the picklist shared component displays the first item in the new picklist BUT keeps the value of the previous picklist, resulting in the incorrect string being displayed for the picklist control. It's not certain whether this is an Angular issue or otherwise but updating the picklist on autofetcher firing has been disabled for now.
The risk of this is: IF someone updates a park/facility and the autofetcher picks up the change, and IF a metrics user has been sitting on the page long enough to pull the updated change, the available parks/facilities to pick from the metrics filter may be out of date (but can easily be fixed with a page refresh).

**Review by meredithom — APPROVED** [review]

**Pull request merged**